django-stubs/views/decorators/__init__.pyi

```diff
@@ -0,0 +1,5 @@
+from typing import TypeVar
+
+from django.utils.deprecation import _AsyncGetResponseCallable, _GetResponseCallable
+
+_ViewFuncT = TypeVar("_ViewFuncT", bound=_AsyncGetResponseCallable | _GetResponseCallable)  # noqa: PYI018
```

Comment:
Would make sense to keep `_AsyncGetResponseCallable`, `_GetResponseCallable` definitions together in the same file with `_ViewFuncT`.

Maybe move those 2 into this file? Or put all of them in `django-stubs/views/__init__.pyi`?

Reply:
Waitaminute. Views aren't necessarily compatible with `_GetResponseCallable`.

Views can have extra arguments from URL parameters, e.g. `def good_view2(request: HttpRequest, url_param: str) -> HttpResponse`

This causes error

> error: Value of type variable "_ViewFuncT" of "csrf_protect" cannot be "Callable[[HttpRequest, str], HttpResponse]"  [type-var]

Reply:
Yep as suggested by `flaeppe`. I got mislead because some of these decorators are defined using `decorator_from_middleware`, which will call `get_response` (i.e. the view function) only by passing the `request`. So probably we need another view func type for those

Reply:
Ok my bad, `decorator_from_middleware` also supports extra arguments specified in a view function. So in the end I only need to update the definition of `_ViewFuncT`.